ENIGMAsystem/SHELL/Widget_Systems/General/tinyfiledialogs/dialogs.cpp

```diff
@@ -287,9 +294,10 @@ double get_integer(string str, double def) {
   else if (caption == "" && tfd_DialogEngine() == tfd_KDialog)
     caption = "KDialog";
 
-  std::ostringstream def_integer;
-  def_integer << def;
-  string integer = def_integer.str();
+  if (def > 999999999999999) 
+    def = 999999999999999;
```

Comment:
What is the significance of this constant? It's too large to fit into an integer, so the compiler is probably treating it as a long. It's smaller than the maximum storage of a long. Based on the number of bits used, this is the maximum number of nines you can fit in a double (without loss of precision), but that's not meaningful in and of itself—nothing is special about the number nine. The biggest number you can store without loss of precision is `2 << 56 - 1`, and there should be a constant for that, but if your solution is to cap it, why bother? That's just a greater loss of precision.

Reply:
@JoshDreamland I wasn't 100% following what you were trying to say. Let me know if my latest commit does what you wanted me to do.